fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -187,18 +222,266 @@ protected Expr uncheckedCastTo(Type targetType) throws AnalysisException {
     @Override
     public void write(DataOutput out) throws IOException {
         super.write(out);
-        out.writeLong(date.getTime());
+        long ymd = ((year * 13 + month) << 5) | day;
+        long hms = (hour << 12) | (minute << 6) | second;
+        long packed_datetime = ((ymd << 17) | hms) << 24 + microsecond;
+        out.writeLong(packed_datetime);
     }
 
     @Override
     public void readFields(DataInput in) throws IOException {
         super.readFields(in);
-        date = new Date(in.readLong());
+        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_59) {
+            long packed_time = in.readLong();
+            microsecond = (packed_time % (1L << 24));
+            long ymdhms = (packed_time >> 24);
+            long ymd = ymdhms >> 17;
+            long hms = ymdhms % (1 << 17);
+
+            day = ymd % (1 << 5);
+            long ym = ymd >> 5;
+            month = ym % 13;
+            year = ym / 13;
+            year %= 10000;
+            second = hms % (1 << 6);
+            minute = (hms >> 6) % (1 << 6);
+            hour = (hms >> 12);
+            this.type = Type.DATETIME;
```

Comment:
why DATETIME? I think it may be DATE

Reply:
The dates read from the meta are all datetime types, and Doris explicitly specifies type DATE/DATETIME by setType()